turtlesim/CMakeLists.txt

```diff
@@ -13,35 +13,55 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(ament_index_cpp REQUIRED)
 find_package(geometry_msgs REQUIRED)
-find_package(Qt5 REQUIRED COMPONENTS Widgets)
+
+if(USE_QT6)
+  find_package(Qt6 REQUIRED COMPONENTS Widgets)
+  set(QT_VERSION_MAJOR 6)
+else()
+  find_package(Qt5 REQUIRED COMPONENTS Widgets)
+  set(QT_VERSION_MAJOR 5)
+endif()
+
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(std_srvs REQUIRED)
 find_package(turtlesim_msgs REQUIRED)
 
-include_directories(include ${Qt5Widgets_INCLUDE_DIRS})
-
-qt5_wrap_cpp(turtlesim_node_MOCS include/turtlesim/turtle_frame.hpp)
+if(${QT_VERSION_MAJOR} GREATER "5")
+  qt_standard_project_setup()
+  qt_wrap_cpp(turtlesim_node_MOCS include/turtlesim/turtle_frame.hpp)
+else()
+  qt5_wrap_cpp(turtlesim_node_MOCS include/turtlesim/turtle_frame.hpp)
+endif()
 
 add_executable(turtlesim_node
   src/turtlesim.cpp
   src/turtle.cpp
   src/turtle_frame.cpp
   ${turtlesim_node_MOCS}
 )
+target_include_directories(turtlesim_node PRIVATE
```

Comment:
this seems odd, does qt not provide a target for this ?

Reply:
qt_wrap_cpp(turtlesim_node_MOCS 
include/turtlesim/turtle_frame.hpp
TARGET MocsTarget
)
target_link_libraries(turtlesim_node PRIVATE MocsTarget)

Does this work ?

Reply:
add_library(MocHeaders INTERFACE)
target_include_directories(MocHeaders INTERFACE
  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
)

Now you can use targetLink libraries below with MocHeaders

Reply:
I don't understand the fix, do you mind to suggest the changes?

Reply:
all the target_include_directories can now be removed